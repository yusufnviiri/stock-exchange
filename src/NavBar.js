import React from 'react';
import { Link } from 'react-router-dom';
import { AiOutlineAudio, AiOutlineSetting } from 'react-icons/ai';

function NavBar() {
  const links = [
    {
      id: 1,
      path: '/',
      text: <AiOutlineSetting className="settingIcon" />,
    },
    {
      id: 2,
      path: '/details',
      text: <AiOutlineAudio className="audioIcon" />,
      class: 'item2',
    },
  ];
  return (
    <nav className="navBar">

      <ul>

        {links.map((link) => (
          <li key={link.id} id={link.class}>
            <Link to={link.path}>{link.text}</Link>
          </li>
        ))}
      </ul>

    </nav>
  );
}
export default NavBar;
