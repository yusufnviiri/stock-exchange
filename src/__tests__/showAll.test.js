/* eslint-disable  */
import React from 'react';
import * as router from 'react-router';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import ShowAll from '../components/showAll';
import StockData from '../components/stockData';
import store from '../redux/Store';
import NavBar from '../NavBar';


describe('Ensure app renders as expected', () => {
    const navigate = jest.fn();  
    beforeEach(() => {
      jest.spyOn(router, 'useNavigate').mockImplementation(() => navigate);
    });
    
    test('renders the Details page', () => {
      render(<Provider store={store}><StockData /></Provider>);
      expect(screen.findAllByText('Price Change:')).toMatchSnapshot();
    });
    test('renders the home page', () => {
        render(<Provider store={store}><ShowAll /></Provider>);
        expect(screen.findAllByText(' Real Time Crypto currency data')).toMatchSnapshot();
      });
   

   
  });
  