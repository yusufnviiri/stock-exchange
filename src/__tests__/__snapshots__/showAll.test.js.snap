// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ensure app renders as expected renders the Details page 1`] = `Promise {}`;

exports[`Ensure app renders as expected renders the home page 1`] = `Promise {}`;
